rustworkx/__init__.py

```diff
@@ -1491,40 +1491,63 @@ def cartesian_product(
 
 @_rustworkx_dispatch
 def bfs_search(graph, source, visitor):
-    """Breadth-first traversal of a directed/undirected graph.
+    """Iterative breadth-first traversal of a directed/undirected graph.
 
-    The pseudo-code for the BFS algorithm is listed below, with the annotated
-    event points, for which the given visitor object will be called with the
-    appropriate method.
+    Pseudo-code for the iterative breadth-first search algorithm is
+    listed below, including the event points for which the given
+    :class:`~rustworkx.visit.BFSVisitor` visitor object will be called with
+    the appropriate method.
 
     ::
 
-        BFS(G, s)
-          for each vertex u in V
-              color[u] := WHITE
-          end for
-          color[s] := GRAY
-          EQUEUE(Q, s)                             discover vertex s
-          while (Q != Ø)
-              u := DEQUEUE(Q)
-              for each vertex v in Adj[u]          (u,v) is a tree edge
-                  if (color[v] = WHITE)
-                      color[v] = GRAY
-                  else                             (u,v) is a non - tree edge
-                      if (color[v] = GRAY)         (u,v) has a gray target
-                          ...
-                      else if (color[v] = BLACK)   (u,v) has a black target
-                          ...
-              end for
-              color[u] := BLACK                    finish vertex u
-          end while
-
-    If an exception is raised inside the callback function, the graph traversal
+        def BFSIterator(G, I, F):
```

Comment:
For Python I think it makes more sense to talk about `BFS` than `BFSIterator`. I'd rewrite it to keep the old signature ` BFS(G, s)`